test/trento/software_updates_test.exs

```diff
@@ -343,6 +343,50 @@ defmodule Trento.SoftwareUpdates.SettingsTest do
                SoftwareUpdates.change_settings(change_submission, Trento.Support.DateService.Mock)
     end
 
+    test "should support idempotent sequential settings updates" do
+      now = DateTime.utc_now()
+
+      expect(
+        Trento.Support.DateService.Mock,
+        :utc_now,
+        fn -> now end
+      )
+
+      %{
+        url: _initial_url,
+        username: initial_username,
+        password: initial_password,
+        ca_cert: _initial_ca_cert,
+        ca_uploaded_at: _initial_ca_uploaded_at
+      } =
+        insert(
+          :software_updates_settings,
+          [ca_cert: Faker.Lorem.sentence(), ca_uploaded_at: DateTime.utc_now()],
+          conflict_target: :id,
+          on_conflict: :replace_all
+        )
+
+      change_submission = %{
+        url: new_url = "https://new.com",
+        ca_cert: new_ca_cert = "new_ca_cert"
+      }
+
+      Enum.each(1..3, fn _ ->
+        assert {:ok,
+                %{
+                  url: ^new_url,
+                  username: ^initial_username,
+                  password: ^initial_password,
+                  ca_cert: ^new_ca_cert,
+                  ca_uploaded_at: ^now
```

Comment:
Wouldn't this check always pass as the `DateService` mock is always going to return `now`, for every run of `change_settings()`?